pcsx2-qt/ShortcutCreationDialog.cpp

```diff
@@ -0,0 +1,118 @@
+	: QDialog(parent)
+	, m_title(title)
+	, m_path(path)
+{
+	m_ui.setupUi(this);
+	this->setWindowTitle(tr("Creating Shortcut For %1").arg(title));
+	this->setWindowIcon(QtHost::GetAppIcon());
+
+	connect(m_ui.overrideBootELFButton, &QPushButton::clicked, [&]() {
+		const QString path = QFileDialog::getOpenFileName(this, tr("Select ELF File"), QString(), tr("ELF Files (*.elf);;All Files (*.*)"));
+		if (!path.isEmpty())
+			m_ui.overrideBootELFPath->setText(path);
+	});
+
+	connect(m_ui.loadStateFileBrowse, &QPushButton::clicked, [&]() {
+		const QString path = QFileDialog::getOpenFileName(this, tr("Select Save State File"), QString(), tr("Save States (*.p2s);;All Files (*.*)"));
+		if (!path.isEmpty())
+			m_ui.loadStateFileToggle->setText(path);
+	});
+
+	connect(m_ui.overrideBootELFToggle, &QCheckBox::toggled, m_ui.overrideBootELFPath, &QLineEdit::setEnabled);
+	connect(m_ui.overrideBootELFToggle, &QCheckBox::toggled, m_ui.overrideBootELFButton, &QPushButton::setEnabled);
+	connect(m_ui.gameArgsToggle, &QCheckBox::toggled, m_ui.gameArgs, &QLineEdit::setEnabled);
+	connect(m_ui.loadStateIndexToggle, &QCheckBox::toggled, m_ui.loadStateIndex, &QSpinBox::setEnabled);
+	connect(m_ui.loadStateFileToggle, &QCheckBox::toggled, m_ui.loadStateFilePath, &QLineEdit::setEnabled);
+	connect(m_ui.loadStateFileToggle, &QCheckBox::toggled, m_ui.loadStateFileBrowse, &QPushButton::setEnabled);
+
+	connect(m_ui.bootOptionToggle, &QCheckBox::toggled, [&](bool checked) {
+		if (checked)
+			m_ui.bootOptionDropdown->setEnabled(true);
+		else
+			m_ui.bootOptionDropdown->setEnabled(false);
+	});
+
+	connect(m_ui.fullscreenMode, &QCheckBox::toggled, [&](bool checked) {
+		if (checked)
+			m_ui.fullscreenModeDropdown->setEnabled(true);
+		else
+			m_ui.fullscreenModeDropdown->setEnabled(false);
+	});
+
+	m_ui.shortcutDesktop->setChecked(true);
+	m_ui.overrideBootELFPath->setEnabled(false);
+	m_ui.overrideBootELFButton->setEnabled(false);
+	m_ui.gameArgs->setEnabled(false);
+	m_ui.bootOptionDropdown->setEnabled(false);
+	m_ui.fullscreenModeDropdown->setEnabled(false);
+	m_ui.loadStateIndex->setEnabled(false);
+	m_ui.loadStateFileBrowse->setEnabled(false);
+	m_ui.loadStateFilePath->setEnabled(false);
+
+	m_ui.customArgsInstruction->setText(tr("You may add additional (space-separated) <a href=\"https://pcsx2.net/docs/post/cli/\">custom arguments</a> that are not listed above here:"));
+	m_ui.customArgsInstruction->setTextFormat(Qt::RichText);
+	m_ui.customArgsInstruction->setTextInteractionFlags(Qt::TextBrowserInteraction);
+	m_ui.customArgsInstruction->setOpenExternalLinks(true);
+
+	connect(m_ui.dialogButtons, &QDialogButtonBox::accepted, this, [&]() {
+		std::string args;
+
+		if (m_ui.portableModeToggle->isChecked())
+			args += "-portable ";
+
+		if (m_ui.overrideBootELFToggle->isChecked() && !m_ui.overrideBootELFPath->text().isEmpty())
+			args += fmt::format("-elf \"{}\" ", m_ui.overrideBootELFPath->text().toStdString());
```

Comment:
Again, there should be logic to escape the arguments here. I know I've gone on about this before, and it sounds like you'd prefer for this PR to be merged and then for it to be fixed later.

As a hold over measure use single quotes here though so you at least don't have to worry about variable substitutions and such.